crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -314,6 +317,9 @@ proc.compute_and_remove_fee
     exec.build_native_fee_asset
     # => [FEE_ASSET]
 
+    emit.EPILOGUE_TX_FEE_COMPUTED
+    # => [FEE_ASSET]
```

Comment:
If we use `asset_vault::peek_balance` from https://github.com/0xMiden/miden-base/pull/1746 before `exec.compute_fee` on line 313 above, we'd be able to arrange the stack here as something like `[FEE_ASSET, CURRENT_NATIVE_ASSET]`. Then, we wouldn't need to keep track of the `initial_native_asset` in the `TransactionExecutorHost` - right?

I guess this would cost us probably extra 50 cycles or so (though I didn't actually estimate this properly) - but at the same time, should let us avoid a lot of extra code complexity.